File: fixtures/scale/dart/large/lib/ciphers/playfair/playfair_suite.dart
import '../../core/cipher.dart';
import '../../core/cipher_suite.dart';
import '../../core/suite_result.dart';
import '../../verify/round_trip.dart';
import 'playfair_cipher.dart';
import 'playfair_key.dart';
import 'playfair_vectors.dart';

class PlayfairSuite implements CipherSuite {
  @override
  String name() => 'playfair';

  @override
  String category() => 'cipher';

  @override
  SuiteResult run() {
    final Cipher cipher = PlayfairCipher(PlayfairKey.defaultKey());
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in playfairSamples()) {
      if (roundTripCheck(cipher, sample)) {
        passed++;
      } else {
        failed++;
      }
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}

File: fixtures/scale/scala/large/src/cryptobench/ciphers/playfair/PlayfairSuite.scala
package cryptobench.ciphers.playfair

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.RoundTrip

final class PlayfairSuite extends CipherSuite {
  override def name: String = "playfair"

  override def run(): SuiteResult = {
    val cipher: Cipher = new PlayfairCipher(PlayfairKey.default())
    var passed = 0
    var failed = 0
    val start = System.nanoTime()
    for (sample <- PlayfairVectors.samples()) {
      if (RoundTrip.check(cipher, sample)) passed += 1 else failed += 1
    }
    SuiteResult(name, passed, failed, System.nanoTime() - start)
  }
}
